apps/storybook/src/TiledHeatmap.stories.tsx

```diff
@@ -0,0 +1,210 @@
+import {
+  getLevelsSize,
+  PanMesh,
+  VisCanvas,
+  ZoomMesh,
+  TiledHeatmap,
+  TilesApi,
+} from '@h5web/lib';
+import type { ColorMap, Domain, Size } from '@h5web/lib';
+import { ScaleType } from '@h5web/shared';
+import type { Meta, Story } from '@storybook/react/types-6-0';
+import { clamp } from 'lodash';
+import ndarray from 'ndarray';
+import { useMemo } from 'react';
+import { createFetchStore } from 'react-suspense-fetch';
+import type { Vector2 } from 'three';
+
+import FillHeight from './decorators/FillHeight';
+
+// See https://en.wikipedia.org/wiki/Mandelbrot_set
+function mandelbrot(
+  iterations: number,
+  xDomain: Domain,
+  yDomain: Domain,
+  size: Size
+) {
+  const array = ndarray(new Float32Array(size.height * size.width), [
+    size.height,
+    size.width,
+  ]);
+  array.data.fill(0);
+
+  for (let row = 0; row < size.height; row += 1) {
+    const cImag =
+      yDomain[0] + (yDomain[1] - yDomain[0]) * (row / (size.height - 1));
+    for (let col = 0; col < size.width; col += 1) {
+      const cReal =
+        xDomain[0] + (xDomain[1] - xDomain[0]) * (col / (size.width - 1));
+      // z = c
+      let zReal = cReal;
+      let zImag = cImag;
+      for (let index = 0; index <= iterations; index += 1) {
+        // z = z**2 + c
+        const zRealSq = zReal ** 2;
+        const zImagSq = zImag ** 2;
+        zImag = 2 * zReal * zImag + cImag;
+        zReal = zRealSq - zImagSq + cReal;
+
+        // check divergence
+        if (zRealSq + zImagSq > 4) {
+          array.set(row, col, index / iterations);
+          break;
+        }
+      }
+    }
+  }
+  return array;
+}
+
+interface TileKey {
+  lod: number;
+  offset: Vector2;
+}
+
+class MandelbrotTilesApi extends TilesApi {
+  public readonly xDomain: Domain;
+  public readonly yDomain: Domain;
+  private readonly store;
+
+  public constructor(
+    size: Size,
+    tileSize: Size,
+    xDomain: Domain,
+    yDomain: Domain
+  ) {
+    super(tileSize, getLevelsSize(size, tileSize));
+    this.xDomain = xDomain;
+    this.yDomain = yDomain;
+
+    this.store = createFetchStore(
```

Comment:
I'm wondering whether the store should be part of the `api` or integrated in the `TiledHeatmap` component.

Reply:
Hmm yeah, I see your point. That's why, in `@h5web/app`, we have a `Provider` that abstracts the creation of the stores with the methods of the `ProviderAPI` implementations. I'm not too keen on using `react-suspense-fetch` in @h5web/lib and creating the stores in `TiledHeatmap`, though, as I feel like caching should remain the responsibility of the consumer (or another package maybe).

Let's leave the store creation in the Storybook for now.